iOS/Flows/Conversation/Cells/MessagesTimeMachineCollectionViewLayout.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  MessageTimeMachineCollectionViewLayout.swift
+//  Jibber
+//
+//  Created by Martin Young on 12/3/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import StreamChat
+
+/// A subclass of the TimeMachineLayout used to display conversation messages.
+/// In addition to normal time machine functionality, this class also adjusts the color, brightness and other message specific attributes
+/// as the items move along the z axis.
+class MessagesTimeMachineCollectionViewLayout: TimeMachineCollectionViewLayout {
+
+    override class var layoutAttributesClass: AnyClass {
+        return ConversationMessageCellLayoutAttributes.self
+    }
+
+    // MARK: - Layout Configuration
+
+    /// How bright the background of the frontmost item is. 0 is black, 1 is full brightness.
+    var frontmostBrightness: CGFloat = 0.89
+    /// How bright the background of the backmost item is. This is based off of the frontmost item brightness.
+    var backmostBrightness: CGFloat {
+        return self.frontmostBrightness - CGFloat(self.stackDepth+1)*0.05
+    }
+
+    override func layoutAttributesForItemAt(indexPath: IndexPath,
+                                            withNormalizedZOffset normalizedZOffset: CGFloat) -> UICollectionViewLayoutAttributes? {
+
```

Comment:
This is a new function that should it make easier to configure time machine cells.

NormalizedZOffset is a value between -1 and 1.
-1 is a cell all the way in the back.
0 is in front and in focus.
1 is scaled all the way up and in the very front.